### PR TITLE
refactor: modify User-Agent header

### PR DESCRIPTION
**Pull request by mojito317**
## PR summary

Normalize `User-Agent` header to match new sdks specs.

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [x] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

An example:
```
cloudant-node-sdk/0.0.25 (node.version=v16.13.1; os.name=darwin; os.version=20.6.0; os.arch=x64; lang=node.js;)
```
On 64 chars:
```
cloudant-node-sdk/0.0.25 (node.version=v16.13.1; os.name=darwi
```

### CODE REVIEW

File: test/unit/common.test.js [hunk 1]
```diff
@@ -20,10 +20,18 @@ const { getSdkHeaders } = common;
 
 describe('Tests of Common Library', () => {
   describe('getSdkHeaders', () => {
-    test('should return correct User-Agent header', () => {
-      const headers = getSdkHeaders('service1', 'v1', 'operation1');
+    test('should return correct X-IBMCloud-SDK-Analytics header', () => {
```
Comment (tomblench) [inline test/unit/common.test.js:23]: minor: this is testing both analytics _and_ user agent headers

Comment (mojito317) [reply test/unit/common.test.js]: Right! I modified it in [e8a8f75](https://github.com/IBM/cloudant-node-sdk/pull/721/commits/e8a8f75cd320c6b1b9c699fac1a27a47ec534e82)

**Review by tomblench — APPROVED** [review]

File: lib/common.ts [hunk 2]
```diff
@@ -35,10 +35,11 @@ export function getSdkHeaders(
   const sdkVersion = pkg.version;
   const osName = os.platform();
   const osVersion = os.release();
+  const osArch = os.arch();
   const nodeVersion = process.version;
 
   const headers = {
-    'User-Agent': `${sdkName}-${sdkVersion} ${osName} ${osVersion} ${nodeVersion}`,
+    'User-Agent': `${sdkName}/${sdkVersion} (node.version=${nodeVersion}; os.name=${osName}; os.version=${osVersion}; os.arch=${osArch}; lang=node.js;)`,
```
Comment (ricellis) [inline lib/common.ts:42]: nit
```suggestion
    'User-Agent': `${sdkName}/${sdkVersion} (node.version=${nodeVersion}; os.name=${osName}; os.version=${osVersion}; os.arch=${osArch}; lang=Node.js;)`,
```

File: package.json [hunk 3]
```diff
@@ -27,7 +27,7 @@
     "test-unit": "npm run build && jest test/unit/",
     "test-integration": "npm run build && jest test/integration",
     "test-travis": "jest --runInBand --testNamePattern='^((?!@slow).)*$' test/",
-    "test-unit-travis": "jest --runInBand test/unit/",
+    "test-unit-travis": "jest --runInBand test/unit/common.test.js",
```
Comment (ricellis) [inline package.json:30]: Unintentionally disabling the other unit tests I think.
```suggestion
    "test-unit-travis": "jest --runInBand test/unit/",
```

**Review by ricellis — CHANGES_REQUESTED** [review]
+1 for the UA change, but it shouldn't be turning off other UTs.

File: package.json (hunk 3 above)
Comment (mojito317) [reply package.json]: [1c5baa7](https://github.com/IBM/cloudant-node-sdk/pull/721/commits/1c5baa7d5144eaab505ff1a1b29cafd888689340)

File: lib/common.ts (hunk 2 above)
Comment (mojito317) [reply lib/common.ts]: [1c5baa7](https://github.com/IBM/cloudant-node-sdk/pull/721/commits/1c5baa7d5144eaab505ff1a1b29cafd888689340)

**Review by ricellis — APPROVED** [review]

**Comment by mojito317** [comment]
Thanks for the reviews @tomblench and @ricellis! I'm about to squash and merge this!

**Pull request merged**